### PR TITLE
fix: Only write Vite config files when they have changed

### PR DESCRIPTION
**Pull request by Artur-**
Prevents Vite from restarting when not needed

Fixes https://github.com/vaadin/hilla/issues/1134

### CODE REVIEW

**Comment by Artur-** [comment]
The problem is clearly seen in Hilla+React projects as any Java change results in
```
2023-08-14T10:03:22.761+03:00  INFO 72515 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   : [vite] target/plugins/application-theme-plugin/theme-generator.js changed, restarting server...
2023-08-14T10:03:22.762+03:00  INFO 72515 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   : [vite] target/plugins/application-theme-plugin/theme-copy.js changed, restarting server...
2023-08-14T10:03:22.763+03:00  INFO 72515 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   : [vite] target/plugins/application-theme-plugin/theme-handle.js changed, restarting server...
2023-08-14T10:03:22.763+03:00  INFO 72515 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   : [vite] target/plugins/theme-loader/theme-loader-utils.js changed, restarting server...
2023-08-14T10:03:22.763+03:00  INFO 72515 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   : [vite] target/plugins/rollup-plugin-postcss-lit-custom/rollup-plugin-postcss-lit.js changed, restarting server...
2023-08-14T10:03:22.817+03:00  INFO 72515 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   : [vite] server restarted.
```
followed by
```
2023-08-14T10:03:23.166+03:00  INFO 72515 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   : [vite] target/vaadin-dev-server-settings.json changed, restarting server...
2023-08-14T10:03:23.166+03:00  INFO 72515 --- [v-server-output] c.v.b.devserver.DevServerOutputTracker   : [vite] vite.generated.ts changed, restarting server...
```

**Comment by Artur-** [comment]
This has not been a problem in earlier version as there was a bug in Vite preventing it from correctly watching all its config files for changes. Now that Vite works, it is an issue

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskInstallFrontendBuildPlugins.java [hunk 1]
```diff
@@ -85,35 +85,27 @@ private void generatePluginFiles(String pluginName) throws IOException {
             return;
         }
 
-        // Validate installed version and don't override if same
```
Comment (caalador) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskInstallFrontendBuildPlugins.java:88]: Can we keep the updated flag as having to build a jar when developing/debugging is a pain.

Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskInstallFrontendBuildPlugins.java:88]: It is back

**Pull request merged**